src/foam/dao/PipelinePMDAO.java

```diff
@@ -0,0 +1,110 @@
+/**
+ * @license
+ * Copyright 2019 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package foam.dao;
+
+import foam.core.FObject;
+import foam.core.X;
+import foam.mlang.order.Comparator;
+import foam.mlang.predicate.Predicate;
+import foam.nanos.pm.PM;
+
+public class PipelinePMDAO
+  extends ProxyDAO
+{
+  protected String putName_;
+  protected String findName_;
+  protected String removeName_;
+  protected String removeAllName_;
+  protected String delegateName_;
+
+  public PipelinePMDAO(X x, DAO delegate) {
```

Comment:
If you do this instead: public PipelinePMDAO(X x, ProxyDAO delegate) {
then can't even be called with a non ProxyDAO.